cohere/responses/chat.py

```diff
@@ -72,39 +72,55 @@ async def respond(self, response: str, max_tokens: int = None) -> "AsyncChat":
         )
 
 
-StreamingText = NamedTuple("StreamingText", [("index", Optional[int]), ("text", str)])
+StreamingText = NamedTuple("StreamingText", [("index", Optional[int]), ("text", str), ("is_finished", bool)])
 
 
 class StreamingChat(CohereObject):
     def __init__(self, response):
         self.response = response
         self.texts = []
-
-    def _make_response_item(self, line) -> Any:
+        self.response_id = None
+        self.conversation_id = None
+        self.preamble = None
+        self.prompt = None
+        self.chatlog = None
+        self.finish_reason = None
+
+    def _make_response_item(self, index, line) -> Any:
         streaming_item = json.loads(line)
-        index = streaming_item.get("index", 0)
+        is_finished = streaming_item.get("is_finished")
         text = streaming_item.get("text")
 
-        while len(self.texts) <= index:
-            self.texts.append("")
+        if not is_finished:
+            return StreamingText(text=text, is_finished=is_finished, index=index)
```

Comment:
I worry that if there's no `is_finished` key then this could exit early, not a big deal if there's' a guarantee that this keys exists.

Reply:
I'll also check for `finish_reason` in the response just in case, but `is_finished` should always exist.

Reply:
Actually, I don't think we need an extra check... Looking at the endpoint, if this key doesn't exist something else will be wrong there